deploy/crds/logging.openshift.io_clusterlogforwarders_crd.yaml

```diff
@@ -32,6 +32,41 @@ spec:
         spec:
           description: ClusterLogForwarderSpec defines the desired state of ClusterLogForwarder
           properties:
+            inputs:
+              description: Inputs are named inputs of log messages
+              items:
+                description: InputSpec defines a source of log messages.
+                properties:
+                  application:
+                    description: Application specific specs
+                    properties:
+                      namespaceNames:
+                        description: "NamespaceNames referes to the namespace names.
+                          The input to the pipeline will be restricted to logs from
+                          containers belonging to these namespaces. \n If no namespaces
+                          are specified, all application container logs will be collected."
+                        items:
+                          type: string
+                        type: array
+                    required:
+                    - namespaceNames
```

Comment:
Make it optional. In future an InputSpec will have other fields for label selectors etc. so you won't always want to specify namespaces. For now an InputSpec with no namespaces is just a long-winded way of referring to the Type, i.e.
```
inputs: [ { name: foo, type: application } ]
pipeline: { inputRefs: [foo] }
```
is the same as
```
pipeline: { inputRefs: [application] }
```

Reply:
updated